app/view/form/LayerTreeFilter.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Panel with filtering tools for the layer tree.
+ *
+ * @class CpsiMapview.view.from.LayerTreeFilter
+ */
+Ext.define('CpsiMapview.view.form.LayerTreeFilter', {
+    extend: 'Ext.form.FieldSet',
+
+    xtype: 'cmv_layertreefilter',
+
+    requires: [
+        'CpsiMapview.controller.form.LayerTreeFilter'
+    ],
+
+    controller: 'cmv_layertreefilter',
+
+    config: {
+        /**
+         * If baselayes shall be filtered by text filter.
+         */
+        doFilterBaseLayers: true
+    },
+
+    /**
+     * The ID of the filter created by the text field.
+     */
+    TEXT_FILTER_ID: 'cpsi-tree-layer-text-filter',
+
+    /**
+     * The ID of the filter created by the checkbox.
+     */
+    VISIBLE_LAYER_FILTER_ID: 'cpsi-tree-visible-layer-filter',
+
+    items: [{
+        xtype: 'textfield',
+        emptyText: 'Search for layers',
+        hideTrigger: true,
+        anchor: '100%',
+        maxWidth: 250,
+        triggers: {
+            clearText: {
```

Comment:
Out of interest - when is this function called?

Reply:
The function is called when the user clicks on the cross. The name `clearText` is just a custom name (see https://docs.sencha.com/extjs/6.2.0/classic/Ext.form.field.Text.html#cfg-triggers). It could be any other name. In fact, with 051afdd I changed its name to `clearTreeFilterText` to make it more expressive.

Reply:
Thanks for the explanation. I hadn't realised `trigger` was the name ExtJS uses to refer to the button associated with the field and was looking for a custom event.